### PR TITLE
nnn: add page

### PR DESCRIPTION
**Pull request by chrillebile**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/nnn.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# nnn
+
+> The missing terminal file manager for X.
+> Read more at https://github.com/jarun/nnn.
+
+- Specify bookmark key to open:
```
Comment (mebeim) [inline pages/common/nnn.md]:
```suggestion
- Open an existing bookmark:
```
I'd say this reads better.

File: pages/common/nnn.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+`nnn -b {{key}}`
```
Comment (mebeim) [inline pages/common/nnn.md]:
```suggestion
`nnn -b {{bookmark_name}}`
```
Let's make it clearer what the parameter means.

File: pages/common/nnn.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+- Start in light mode (fewer details):
+
+`nnn -l`
+
+- Use version compare to sort files:
+
+`nnn -n`
+
+- Copy (or pick) selection to file, or stdout if file='-':
```
Comment (mebeim) [inline pages/common/nnn.md]: I'm not really sure what this command does, could you provide an example usage? It does not seem to do what's described here (at least for what I tried).

File: pages/common/nnn.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+`nnn -p {{file}}`
+
+- Use substring match for filters instead of regex:
```
Comment (mebeim) [inline pages/common/nnn.md]:
```suggestion
- Use substring matching for filters instead of regex:
```

File: pages/common/nnn.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# nnn
+
+> Interactive terminal file manager and fisk analyzer.
```
Comment (mebeim) [inline pages/common/nnn.md]:
```suggestion
> Interactive terminal file manager and disk usage analyzer.
```
Whoops, a typo (also, I forgot the important word "usage", since it's not a _disk analyzer_).

File: pages/common/nnn.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+> More information: <https://github.com/jarun/nnn>.
+
+- Open file manager at current directory, optional: at given directory from current:
```
Comment (mebeim) [inline pages/common/nnn.md]:
```suggestion
- Open the current directory (or specify one as the first argument):
```
This doesn't feel quite right. I'd reword like this.

File: pages/common/nnn.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+`nnn {{path/to/directory}}`
```
Comment (mebeim) [inline pages/common/nnn.md]:
```suggestion
`nnn`
```
Typically the first example is always the *simplest* usage of the command.

File: pages/common/nnn.md (hunk 3 above)
Comment (chrillebile) [reply pages/common/nnn.md]: It just copies the selected file path and outputs it to stdout or a specifyd file.

try: `nnn -p -` and then select an file and press enter

Comment (chrillebile) [reply pages/common/nnn.md]: example of commando + output after I've selected the file "7zr.md"

image

File: pages/common/nnn.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# nnn
+
+> Interactive terminal file manager and disk usage nalyzer.
```
Comment (agnivade) [inline pages/common/nnn.md]: typo at nalyzer

File: pages/common/nnn.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+`nnn -S`
+
+- Start in light mode (fewer details):
+
+`nnn -l`
+
+- Copy selected file path to file_out, or stdout if file_out='-':
+
+`nnn -p {{file_out}}`
```
Comment (agnivade) [inline pages/common/nnn.md]: Just say file_path instead of file_out. And maybe change the latter part of the description to - "or stdout if file_path is '-'"

File: pages/common/nnn.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+- Use substring matching for filters instead of regex:
+
+`nnn -s`
+
+- Use version compare to sort files:
```
Comment (agnivade) [inline pages/common/nnn.md]: This is not clear to me. From where is it getting the version ?

File: pages/common/nnn.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
+# nnn
+
+> Interactive terminal file manager and disk usage nalyzer.
+> More information: <https://github.com/jarun/nnn>.
+
+- Open the current directory (or specify one as the first argument):
+
+`nnn`
+
+- Open an existing bookmark:
```
Comment (agnivade) [inline pages/common/nnn.md]: How do we create a bookmark in the first place ? Maybe this isn't that important compared to other options ? If so, then let us leave it out.

File: pages/common/nnn.md (hunk 10 above)
Comment (mebeim) [reply pages/common/nnn.md]: Good point, at a quick look from building the command I thought it was meant for ordering files based on the version string in their names, for example `abc-1.10` vs `abc-1.2`, but _it does not look like so_ (order should be `1.1`, `1.2`, `1.10` in such case):

    $ touch abc-1.1
    $ touch abc-1.2
    $ touch abc-1.10
    $ nnn

    2019-06-07 12:43 644      0B abc-1.1
    2019-06-07 12:43 644      0B abc-1.10
    2019-06-07 12:43 644      0B abc-1.2

    $ nnn -s

    2019-06-07 12:43 644      0B abc-1.1
    2019-06-07 12:43 644      0B abc-1.10
    2019-06-07 12:43 644      0B abc-1.2

Some explanation would be helpful indeed :thinking:

File: pages/common/nnn.md (hunk 11 above)
Comment (mebeim) [reply pages/common/nnn.md]: Bookmarks are defined in the `NNN_BMS` environment variable, for example in your `.bashrc` you could add:

    export NNN_BMS='d:~/Documents;m:/home/user/Music;c:~/Downloads/Google Chrome'

File: pages/common/nnn.md (hunk 9 above)
Comment (mebeim) [reply pages/common/nnn.md]: I'd suggest the following:

    - Copy the selected file path to a specific output file (stdout if `out_file` is `-`):

    `nnn -p {{file_out}}`

And would also suggest leaving `file_out` as is, since the path being copied is not the one of that file, but the one selected by the user.

Comment (mebeim) [reply pages/common/nnn.md]: Example:

Peek 2019-06-07 12-58

File: pages/common/nnn.md (hunk 3 above)
Comment (mebeim) [reply pages/common/nnn.md]: @chrillebile thank you for the explanation!

File: pages/common/nnn.md (hunk 9 above)
Comment (agnivade) [reply pages/common/nnn.md]: Oh yeah. Then we should follow the usual convention I think - path/to/output_file. Hows that sound ?

File: pages/common/nnn.md (hunk 11 above)
Comment (agnivade) [reply pages/common/nnn.md]: Hmm .. needs extra explanation. I think we can leave this one out.

File: pages/common/nnn.md (hunk 9 above)
Comment (mebeim) [reply pages/common/nnn.md]: @agnivade you're right (edited my comment above just to avoid duplicating code).

File: pages/common/nnn.md (hunk 11 above)
Comment (mebeim) [reply pages/common/nnn.md]: I think this example is ok to have, after all it's probably one of the most used and useful ones. I would suggest editing its description to:

    - Open an existing bookmark (defined in the `NNN_BMS` environment variable):

Comment (agnivade) [reply pages/common/nnn.md]: Sure

File: pages/common/nnn.md (hunk 10 above)
Comment (agnivade) [reply pages/common/nnn.md]: I don't know. We need clarity on how the sorting is done in the first place. If someone can explain that, that would be great.

Comment (sbrl) [reply pages/common/nnn.md]: To me, that looks like a string-based comparison sort - i.e. it compares each character one at a time. On the flip-side, a _natural_ sort would look like this:

```
1.1
1.2
1.10
```

This uses the terminology used in the PHP documentation.

Comment (mebeim) [reply pages/common/nnn.md]: @sbrl yes, we came to the same conclusion. What we were trying to understand was what actually is the difference between `nnn` and `nnn -s`...

Comment (sbrl) [reply pages/common/nnn.md]: Ah, I see. Is there a manual that we can refer to at all, or a developer we can @mention and ask?

Comment (mebeim) [reply pages/common/nnn.md]: This is the GH project, https://github.com/jarun/nnn 

Maybe @jarun could be so kind to help us figure this out?

Comment (jarun) [reply pages/common/nnn.md]: Hey guys,

Thanks for considering `nnn`. Please take a look at the program help from master:

```
usage: nnn [-a] [-b key] [-d] [-f] [-H] [-i] [-n] [-o]
           [-p file] [-r] [-s] [-S] [-t] [-v] [-h] [PATH]

The missing terminal file manager for X.

positional args:
  PATH   start dir [default: current dir]

optional args:
 -a      use access time
 -b key  open bookmark key
 -d      detail mode
 -f      run filter as cmd on prompt key
 -H      show hidden files
 -i      nav-as-you-type mode
 -n      version sort
 -o      press Enter to open files
 -p file selection file (stdout if '-')
 -r      show cp, mv progress on Linux
 -s      string filters [default: regex]
 -S      du mode
 -t      disable dir auto-select
 -v      show version
 -h      show help
```

I am suggesting master because there would be some changes in the upcoming release.

The switch to version sort is `-n`. The option `-s` is to use plain text strings instead of regex at the filter prompt.

**Comment by jarun** [comment]
Please update to nnn v2.8 which is the latest.

**Comment by jarun** [comment]
Also, @pxgamer please let us know if you think this doesn't belong to tldr; (it's been a while). In that case we can close the PR.

**Comment by agnivade** [comment]
Let's just close it for now. @chrillebile, please feel to reopen this whenever you get to it. Thanks.
